### ISSUE TITLE
Type instantiation is excessively deep and possibly infinite. 

### ISSUE THREAD

**Issue by ndrxwsh** [Ayub-Begimkulov/i18n#4]
hello faced with such a problem with my nextjs13 project and experimental app directory. 
![image](https://user-images.githubusercontent.com/94861486/230892182-daacd0a1-4241-45d3-bc37-bbcaa9537625.png)

**Comment by Ayub-Begimkulov** [comment]
Hey Andrew, thanks for reporting this issue. Could please share your whole `I18N` setup and keys that you pass to it?

Cause from this screen I can't really now what might be the problem and how to reproduce it.

**Comment by ndrxwsh** [comment]
Okay, here is my provider in custom component, where i get the error
![image](https://user-images.githubusercontent.com/94861486/230901333-d257468c-16d7-444a-a18b-0c2d062fd7f5.png)
here is setup i18n exactly as in the documentation
![image](https://user-images.githubusercontent.com/94861486/230901503-66d03043-5a86-495c-84c8-0e941910a103.png)
en keys
![image](https://user-images.githubusercontent.com/94861486/230901684-fd9d1582-cade-4755-84bd-2e3513b9798f.png)
ru keys
![image](https://user-images.githubusercontent.com/94861486/230901722-3e47f796-2bce-442a-b7ce-37479a8d98a5.png)
usually error occurred in the hook "useTranslateBase"
or in the I18NProvider

**Comment by Ayub-Begimkulov** [comment]
Yeah, I was able to reproduce this error. I'll look into it more deeply and try to fix tomorrow.

Thanks you very much for reporting an issue!

**Comment by Ayub-Begimkulov** [comment]
Seems like with current API and parameters inferring that the library has, it's not possible to fix this issue. It most likely will require some changes not only in typings, but the API itself also. 

I'll rollback latest inferring improvements and will try to come up with a better solution to this problem.

**Comment by Ayub-Begimkulov** [comment]
I just reverted `inferParameterTypes` option so that we avoid recursive type checking for now. Will try to think of a better solution, but for now no parameters inferring.
